frontend/src/lib/MetricsManager.ts

```diff
@@ -210,4 +217,27 @@ export class MetricsManager {
   private track(evName: string, data: Record<string, unknown>): void {
     analytics.track(evName, data)
   }
+
+  // Get the installation IDs from the session
+  private static getInstallationData(): Record<string, unknown> {
+    return {
+      machineIdV1: SessionInfo.current.installationIdV1,
+      machineIdV2: SessionInfo.current.installationIdV2,
+    }
+  }
+
+  // Use the tracking data injected by S4A if the app is hosted there
+  private static getHostTrackingData(): Record<string, unknown> {
+    if (isInChildFrame() && window.parent.streamlitShareMetadata) {
+      return pick(window.parent.streamlitShareMetadata, [
+        "hosted",
+        "owner",
+        "repo",
+        "branch",
+        "mainModule",
+        "creatorId",
```

Comment:
Just checking: on the metrics side we also have access to the app's URL, right? Or is that something we need to add to streamlitShareMetadata?

Reply:
In BigQuery we have the app's URL

<img width="849" alt="Screen Shot 2020-10-08 at 3 29 41 PM" src="https://user-images.githubusercontent.com/699897/95504658-18679100-097b-11eb-92b7-2e1f79274738.png">

Reply:
<img width="1244" alt="Screen Shot 2020-10-08 at 3 30 55 PM" src="https://user-images.githubusercontent.com/699897/95504779-43ea7b80-097b-11eb-9e3c-ccb876fcf99b.png">